.github/workflows/update_version.yml

```diff
@@ -20,54 +20,36 @@ jobs:
       - name: Get current date for the first release candidate
         if: github.event_name == 'schedule'
         run: |
-          echo "VERSION=$(date +'v%y.%-m')rc0" >> $GITHUB_ENV
-          echo "NON_RC_VERSION=$(date +'v%y.%-m')" >> $GITHUB_ENV
+          echo "VERSION=$(date +'v%y.%-m' + ".0")" >> $GITHUB_ENV
 
       - name: Get current date for a manual release
         if: github.event_name == 'workflow_dispatch'
         run: |
           echo "VERSION=$(date +'v%y.%-m')${{ github.event.inputs.append_to_tag }}" >> $GITHUB_ENV
-          echo "NON_RC_VERSION=$(date +'v%y.%-m')" >> $GITHUB_ENV
 
       # the schedule workflow is for rc0 release
       - uses: actions/checkout@v4
-        if: github.event_name == 'schedule'
         with:
           ref: 'develop'
 
-      # the dispatch workflow is for rcX and final releases
-      - uses: actions/checkout@v4
-        if: github.event_name == 'workflow_dispatch'
-        with:
-          ref: '${{ env.NON_RC_VERSION }}'
-
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
           python-version: 3.12
 
       - name: Install dependencies
         run: |
-          pip install wheel
-          pip install --editable ".[all]"
+          pip install -e ".[all]"
 
       # update all the version strings and add CHANGELOG headings
       - name: Update version
         run: python scripts/update_version.py
 
       # create a new version branch for rc0 release and commit
       - uses: EndBug/add-and-commit@v9
-        if: github.event_name == 'schedule'
-        with:
-          message: 'Bump to ${{ env.VERSION }}'
-          new_branch: '${{ env.NON_RC_VERSION }}'
-
-      # use the already created release branch for rcX + final releases
-      # and commit
-      - uses: EndBug/add-and-commit@v9
-        if: github.event_name == 'workflow_dispatch'
         with:
           message: 'Bump to ${{ env.VERSION }}'
+          new_branch: '${{ env.VERSION }}'
```

Comment:
Creating a new branch for each release will bring in the non-bug-fix commits from develop too. Do we want bug fix releases to have features in them too? The version scheme gets very hazy here.